shell-ui/src/NavBar.js

```diff
@@ -2,64 +2,95 @@
 import CoreUINavbar from '@scality/core-ui/dist/components/navbar/Navbar.component';
 import { useAuth } from 'oidc-react';
 import { useLayoutEffect, useState } from 'react';
-import type { SolutionsNavbarProps } from './index';
+import type {
+  Options,
+  SolutionsNavbarProps,
+  TranslationAndGroups,
+} from './index';
 import type { Node } from 'react';
 import { logOut } from './auth/logout';
+import {
+  getAccessiblePathsFromOptions,
+  isEntryAccessibleByTheUser,
+  normalizePath,
+} from './auth/permissionUtils';
+import { prefetch } from "quicklink";
 
-export const LoadingNavbar = (): Node => <CoreUINavbar role="navigation" tabs={[{title: 'loading'}]} />;
+export const LoadingNavbar = (): Node => (
+  <CoreUINavbar role="navigation" tabs={[{ title: 'loading' }]} />
+);
 
-export const Navbar = ({
-  options,
-}: {
-  options: $PropertyType<SolutionsNavbarProps, 'options'>,
-}): Node => {
+const translateOptionsToMenu = (
+  options: Options,
+  entries: 'main' | 'subLogin',
+  renderer: (
+    path: string,
+    translationAndGroup: TranslationAndGroups,
+  ) => { link: Node } | { label: string, onClick: () => void },
+  userGroups: string[],
+) => {
+  const normalizedLocation = normalizePath(location.href);
+  return (
+    Object.entries(options[entries])
+      //$FlowIssue - flow typing for Object.entries incorrectly typing values as [string, mixed] instead of [string, TranslationAndGroups]
+      .filter((entry: [string, TranslationAndGroups]) =>
+        isEntryAccessibleByTheUser(entry, userGroups),
+      )
+      .map(
+        //$FlowIssue - flow typing for Object.entries incorrectly typing values as [string, mixed] instead of [string, TranslationAndGroups]
+        ([path, translationAndGroup]: [string, TranslationAndGroups], i) => {
+          try {
+            return {
+              ...renderer(path, translationAndGroup),
+              selected: normalizedLocation === normalizePath(path),
+            };
+          } catch (e) {
+            throw new Error(
+              `[navbar][config] Options path should be provided thanks to fully qualified urls such as {protocol}://{host}{path}?{queryParams} but got "${path}" in ${entries}[${i}] `,
```

Comment:
Three comments:

- isn't Prettier complaining about the line-length? I don't like scrolling horizontally :stuck_out_tongue_winking_eye: 
- wording can be improved:

```javascript
throw new Error(
  `[navbar][config] Invalid path specified in "options.${section}": "${path}" `
  + '(keys must be defined as fully qualified URLs, '
  + 'such as "{protocol}://{host}{path}?{queryParams}")'
)
```

- showing the index of an entry in an object isn't really useful, right? Or are JS objects ordered somehow?